### PR TITLE
[Material] Updated icons and fonts

### LINKED ISSUE #18384: How to use alternative themes of material icons
Google announced a bunch of updates to Material Guidelines during io2018. One of the updates is themes of material icons - https://material.io/tools/icons/
It now has "Filled", "Outlined", "Rounded", "Two-Tone",  and "Sharp".

In Flutter, I know how to use regular icons which are "Filled" ones. I also know how I can add individual assets to the `assets` folder in Flutter app and use those in my code but I'd like if they were readily available so that you could use them like -
``` Dart
IconButton(
  icon: Icon(Icons.volume_up), // <=== Point of interest here
  tooltip: 'Increase volume by 10%',
  onPressed: () { setState(() { _volume *= 1.1; }); },
)
```

Looking at [this repo](https://github.com/flutter/material_icons_font) I think it only supports default (Filled) right now but are there any plans to add other themes?

Probably
``` Dart
IconButton(
  icon: Icon(Icons.volume_up, theme: IconTheme.outlined), // <=== Example
  tooltip: 'Increase volume by 10%',
  onPressed: () { setState(() { _volume *= 1.1; }); },
)
```

### LINKED ISSUE #27956: Missing Material Icons
I noticed that some material icons are missing from Flutter including `bookmarks`, `file_copy` and `ballot`.
```
[√] Flutter (Channel beta, v1.1.8, on Microsoft Windows [Version 10.0.17134.523], locale en-US)
    • Flutter version 1.1.8 at C:\src\flutter
    • Framework revision 985ccb6d14 (5 weeks ago), 2019-01-08 13:45:55 -0800
    • Engine revision 7112b72cc2
    • Dart version 2.1.1 (build 2.1.1-dev.0.1 ec86471ccc)

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\dcole\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = C:\Users\dcole\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[√] Android Studio (version 3.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 32.0.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[√] IntelliJ IDEA Community Edition (version 2018.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2018.2.3
    • Flutter plugin version 30.0.2
    • Dart plugin version 182.4999

[√] Connected device (1 available)
    • TA 1039 • PLEGAR2791811777 • android-arm64 • Android 8.1.0 (API 27)
```

### LINKED ISSUE #31665: Request: Add a LogOut Icon to the list of icons
I wanted to add a login/logout button to my application I was surprised to see there is no dedicated icon for logout.

https://docs.flutter.io/flutter/material/Icons-class.html

I don't know what ['Exit_to_app'](https://docs.flutter.io/flutter/material/Icons/exit_to_app-constant.html) is meant to mean, but it definitely isn't right and probably named incorrectly, the icon looks too similar to ['input'](https://docs.flutter.io/flutter/material/Icons/input-constant.html)

The only ones left are ['eject'](https://docs.flutter.io/flutter/material/Icons/eject-constant.html) or  ['phonelink_erase'](https://docs.flutter.io/flutter/material/Icons/phonelink_erase-constant.html) which is what I've settled on.

Thanks

### LINKED ISSUE #43747: Icons.drag_indicator is not available
https://material.io/resources/icons/?icon=drag_indicator&style=baseline

This icon is very useful to have in web apps with resizable components.

### LINKED ISSUE #49014: Add Icons.person_subtract
In a social app, following and unfollowing are both primary actions. Flutter has `Icons.person_add` but doesn't have `Icons.person_subtract`. Adding a `Icons.person_subtract` icon would be useful.

### LINKED ISSUE #52713: Icons.airplanemode_active isn't symetric
I have an eye for detail, if something is out of place it screams to me :sweat_smile:, `Icons.airplanemode_active` isn't symetric:

image

As can be seen, `Icons.airplanemode_active` is a bit to the left, thus any line placed on its center appears to be on its right. This can be confirmed by the distances of the wing tips to the box enclosing the icon:

image

### LINKED ISSUE #61777: Building iOS flutter Error, Cycle inside Runner
I recently upgrade my flutter app to v1.17.5 and getting the below error when trying to run or build the iOS flutter application. I'm not sure whether I should switch the order of some script in the build phase. Any help is appreciated! Thanks

Screen Shot 2020-07-16 at 9 33 38 PM

```
error: Cycle inside Runner; building could produce unreliable results. This usually can be resolved by moving the target's Headers build phase before Compile Sources.
    Cycle details:
    → Target 'Runner': LinkStoryboards
    ○ Target 'Runner' has compile command with input '/Users/ivanchau/AndroidStudioProjects/fella/ios/Runner/Base.lproj/LaunchScreen.storyboard'
    ○ Target 'Runner': Ditto /Users/ivanchau/AndroidStudioProjects/fella/build/ios/Debug-development-iphonesimulator/Runner.swiftmodule/Project/x86_64-apple-ios-simulator.swiftsourceinfo /Users/ivanchau/Library/Developer/Xcode/DerivedData/Runner-gsqfvskdiydqcdftydqhgqbekwhx/Build/Intermediates.noindex/Runner.build/Debug-development-iphonesimulator/Runner.build/Objects-normal/x86_64/Runner.swiftsourceinfo
    ○ Target 'Runner' has compile command for Swift source files
    ○ That command depends on command in Target 'Runner': script phase “Flutter”
    ○ Target 'Runner' has process command with output '/Users/ivanchau/AndroidStudioProjects/fella/build/ios/Debug-development-iphonesimulator/Runner.app/Info.plist'
    ○ Target 'Runner' has compile command with input '/Users/ivanchau/AndroidStudioProjects/fella/ios/Runner/Base.lproj/LaunchScreen.storyboard'
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: Mapping architecture armv7 to i386. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')
    warning: Mapping architecture arm64 to x86_64. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')
```

<details>
  <summary>Logs</summary>

```
 info • Unused import: 'package:font_awesome_flutter/font_awesome_flutter.dart' • lib/components/RoundTextField.dart:2:8 • unused_import
   info • 'title' is deprecated and shouldn't be used. This is the term used in the 2014 version of material design. The modern term is headline6. This feature was deprecated after v1.13.8. •
          lib/components/backdrop.dart:166:39 • deprecated_member_use
   info • The member 'lerp' can only be used within instance members of subclasses of 'package:flutter/src/animation/tween.dart' • lib/components/backdrop.dart:304:59 •
          invalid_use_of_protected_member
   info • Unused import: 'package:fella/style.dart' • lib/components/button/roundButton.dart:2:8 • unused_import
   info • Unused import: 'package:fella/constants.dart' • lib/components/card/showSurveyCard.dart:2:8 • unused_import
   info • Unused import: 'package:fella/layouts/survey/surveyVote.dart' • lib/components/card/showSurveyCard.dart:4:8 • unused_import
   info • Unused import: 'package:fella/layouts/survey/surveyVote.dart' • lib/components/card/surveyCard.dart:4:8 • unused_import
   info • Unused import: 'package:fella/components/imageBox.dart' • lib/components/card/systemAnnouncementCard.dart:3:8 • unused_import
   info • Unused import: 'package:fella/components/button/roundButton.dart' • lib/components/card/systemAnnouncementCard.dart:4:8 • unused_import
   info • Unused import: 'package:fella/data/viewmodels/groupModel.dart' • lib/components/card/systemAnnouncementCard.dart:6:8 • unused_import
   info • Unused import: 'package:provider/provider.dart' • lib/components/card/systemAnnouncementCard.dart:8:8 • unused_import
   info • Don't import implementation files from another package • lib/components/charts/baseCharts/customCircleSymbolRenderer.dart:11:8 • implementation_imports
   info • Don't import implementation files from another package • lib/components/charts/baseCharts/customCircleSymbolRenderer.dart:12:8 • implementation_imports
   info • This class (or a class that this class inherits from) is marked as '@immutable', but one or more of its instance fields aren't final: NumberRow.title, NumberRow.value, NumberRow.min,
          NumberRow.updateVal • lib/components/form/NumberRow.dart:6:7 • must_be_immutable
   info • This class (or a class that this class inherits from) is marked as '@immutable', but one or more of its instance fields aren't final: LoginTextField.label, LoginTextField.icon,
          LoginTextField.obscure, LoginTextField.textInputType, LoginTextField.controller • lib/components/loginTextField.dart:3:7 • must_be_immutable
   info • 'YearPicker' is deprecated and shouldn't be used. Use CalendarDatePicker instead. This feature was deprecated after v1.15.3. • lib/components/picker.dart:90:21 • deprecated_member_use
   info • 'body1' is deprecated and shouldn't be used. This is the term used in the 2014 version of material design. The modern term is bodyText2. This feature was deprecated after v1.13.8. •
          lib/components/picker.dart:183:56 • deprecated_member_use
   info • 'subhead' is deprecated and shouldn't be used. This is the term used in the 2014 version of material design. The modern term is subtitle1. This feature was deprecated after v1.13.8. •
          lib/components/row/titleRow.dart:41:56 • deprecated_member_use
   info • Name non-constant identifiers using lowerCamelCase • lib/constants.dart:95:22 • non_constant_identifier_names
   info • Name non-constant identifiers using lowerCamelCase • lib/constants.dart:104:35 • non_constant_identifier_names
   info • Unused import: 'package:flutter/material.dart' • lib/customClass/userListItem.dart:2:8 • unused_import
   info • The exception variable 'e' isn't used, so the 'catch' clause can be removed • lib/data/calendar.dart:19:33 • unused_catch_clause
   info • Unused import: 'package:fella/data/firebaseObj/survey.dart' • lib/data/services/api.dart:9:8 • unused_import
   info • Unused import: 'dart:collection' • lib/data/services/api/statistics.dart:2:8 • unused_import
   info • Cancel instances of dart.async.StreamSubscription • lib/data/viewmodels/eventModel.dart:17:22 • cancel_subscriptions
   info • Avoid empty statements • lib/data/viewmodels/groupModel.dart:292:5 • empty_statements
   info • Dead code • lib/data/viewmodels/groupModel.dart:292:5 • dead_code
   info • Unused import: 'package:fella/constants.dart' • lib/data/viewmodels/userModel.dart:5:8 • unused_import
   info • Unused import: 'package:firebase_storage/firebase_storage.dart' • lib/data/viewmodels/userModel.dart:13:8 • unused_import
   info • Unused import: 'package:fella/constants.dart' • lib/layoutBase/baseListGroupSelectLayout.dart:2:8 • unused_import
   info • This method overrides a method annotated as '@mustCallSuper' in 'AutomaticKeepAliveClientMixin', but doesn't invoke the overridden method • lib/layouts/dashboard/dashboard.dart:232:10 •
          must_call_super
   info • 'subhead' is deprecated and shouldn't be used. This is the term used in the 2014 version of material design. The modern term is subtitle1. This feature was deprecated after v1.13.8. •
          lib/layouts/dashboard/leaderPage.dart:116:58 • deprecated_member_use
   info • This method overrides a method annotated as '@mustCallSuper' in 'AutomaticKeepAliveClientMixin', but doesn't invoke the overridden method • lib/layouts/dashboard/leaderPage.dart:534:10 •
          must_call_super
   info • Unused import: 'package:fella/style.dart' • lib/layouts/dashboard/submit.dart:25:8 • unused_import
   info • The target expression uses '?.', so its value can be null • lib/layouts/dashboard/tablePage.dart:168:17 • can_be_null_after_null_aware
   info • Unused import: 'package:fella/data/viewmodels/groupModel.dart' • lib/layouts/event/eventMgt.dart:7:8 • unused_import
   info • Unused import: 'package:cached_network_image/cached_network_image.dart' • lib/layouts/event/eventMgt.dart:10:8 • unused_import
   info • Unused import: 'package:fella/layoutBase/baseListLayout.dart' • lib/layouts/event/eventMgt.dart:13:8 • unused_import
   info • Unused import: 'package:transparent_image/transparent_image.dart' • lib/layouts/event/eventMgt.dart:16:8 • unused_import
   info • Unused import: 'package:flutter_linkify/flutter_linkify.dart' • lib/layouts/event/eventPage.dart:24:8 • unused_import
   info • 'subhead' is deprecated and shouldn't be used. This is the term used in the 2014 version of material design. The modern term is subtitle1. This feature was deprecated after v1.13.8. •
          lib/layouts/group/formEdit.dart:62:52 • deprecated_member_use
   info • Unused import: 'package:fella/components/row/customListTile.dart' • lib/layouts/group/groupEdit.dart:7:8 • unused_import
   info • Unused import: 'package:fella/layouts/group/formEdit.dart' • lib/layouts/group/groupEdit.dart:14:8 • unused_import
   info • Unused import: 'package:fella/layouts/group/roleEdit.dart' • lib/layouts/group/groupEdit.dart:15:8 • unused_import
   info • 'subhead' is deprecated and shouldn't be used. This is the term used in the 2014 version of material design. The modern term is subtitle1. This feature was deprecated after v1.13.8. •
          lib/layouts/group/roleEdit.dart:94:56 • deprecated_member_use
   info • This method overrides a method annotated as '@mustCallSuper' in 'AutomaticKeepAliveClientMixin', but doesn't invoke the overridden method • lib/layouts/home/homePage.dart:177:10 •
          must_call_super
   info • Unused import: 'package:firebase_storage/firebase_storage.dart' • lib/layouts/home/infoPage.dart:6:8 • unused_import
   info • This class (or a class that this class inherits from) is marked as '@immutable', but one or more of its instance fields aren't final: ActivatePage.email, ActivatePage.password •
          lib/layouts/login/activatePage.dart:12:7 • must_be_immutable
   info • This method overrides a method annotated as '@mustCallSuper' in 'AutomaticKeepAliveClientMixin', but doesn't invoke the overridden method • lib/layouts/profile/profile.dart:370:10 •
          must_call_super
   info • Unused import: 'package:fella/layouts/group/roleEdit.dart' • lib/layouts/survey/surveyEdit.dart:15:8 • unused_import
   info • Unused import: 'package:fella/components/statText.dart' • lib/layouts/survey/surveyManagement.dart:7:8 • unused_import
   info • Unused import: 'dart:collection' • lib/layouts/survey/surveyResults.dart:1:8 • unused_import
   info • Unused import: 'dart:math' • lib/layouts/survey/surveyResults.dart:3:8 • unused_import
   info • Unused import: 'package:fella/components/button/roundButton.dart' • lib/layouts/survey/surveyResults.dart:5:8 • unused_import
   info • Unused import: 'package:fella/components/row/userRow.dart' • lib/layouts/survey/surveyResults.dart:9:8 • unused_import
   info • Unused import: 'package:fella/data/firebaseObj/user.dart' • lib/layouts/survey/surveyResults.dart:14:8 • unused_import
   info • Unused import: 'package:fella/data/viewmodels/userModel.dart' • lib/layouts/survey/surveyResults.dart:17:8 • unused_import
   info • Unused import: 'package:fella/util.dart' • lib/layouts/survey/surveyResults.dart:19:8 • unused_import
   info • 'subhead' is deprecated and shouldn't be used. This is the term used in the 2014 version of material design. The modern term is subtitle1. This feature was deprecated after v1.13.8. •
          lib/layouts/userForm/contactComponents.dart:40:50 • deprecated_member_use
   info • This method overrides a method annotated as '@mustCallSuper' in 'AutomaticKeepAliveClientMixin', but doesn't invoke the overridden method • lib/layouts/userForm/userPermission.dart:329:10
          • must_call_super
   info • This method overrides a method annotated as '@mustCallSuper' in 'AutomaticKeepAliveClientMixin', but doesn't invoke the overridden method • lib/layouts/userForm/userStatistics.dart:78:10
          • must_call_super
   info • Unused import: 'dart:math' • lib/layouts/userMgt/adminPage.dart:1:8 • unused_import
   info • Unused import: 'package:fella/style.dart' • lib/layouts/userMgt/adminPage.dart:41:8 • unused_import
   info • Unused import: 'package:fella/data/viewmodels/userModel.dart' • lib/layouts/userMgt/emailSettings.dart:2:8 • unused_import
   info • Unused import: 'dart:io' • lib/layouts/userMgt/newGroup.dart:1:8 • unused_import
   info • Unused import: 'dart:math' • lib/layouts/userMgt/roleManagement.dart:2:8 • unused_import
   info • Unused import: 'dart:io' • lib/layouts/userMgt/selectMembers.dart:1:8 • unused_import
   info • Unused import: 'package:fella/style.dart' • lib/layouts/userMgt/selectMembers.dart:11:8 • unused_import
```

```
[✓] Flutter (Channel stable, v1.17.5, on Mac OS X 10.15.4 19E287, locale en-CA)
    • Flutter version 1.17.5 at /Users/ivanchau/Documents/flutter
    • Framework revision 8af6b2f038 (2 weeks ago), 2020-06-30 12:53:55 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/ivanchau/Library/Android/sdk
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = /Users/ivanchau/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.1

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)
```

```
```

</details>

### PR DESCRIPTION
**Pull request by willlarche**
## Description

Additional icons, new icon styles (outline, rounded, sharp), and the latest Roboto and RobotoCondensed fonts. `update_icons.dart` has been updated for additional names and to prevent regressions.

Icon tree shaking has landed for iOS and Android (#56633). There is no icon tree shaking on web (#57181) so there will be an increase in bundle size for that platform.

NOTE: The fonts repo has changed a bit. The wiki on updating material fonts needs to be updated as well as the go link for the internal hosting of the icons font which is now being custom made for us by the Google Fonts team. (Thank you!)

Screen Shot 2020-07-17 at 11 35 31 PM
Screen Shot 2020-07-17 at 11 35 12 PM
Screen Shot 2020-07-17 at 11 35 56 PM
Screen Shot 2020-07-17 at 11 36 14 PM

## Related Issues

fixes #52713
fixes #43747
fixes #31665
fixes #61777 
fixes #27956
fixes #18384
fixes #49014

## Tests

I added the following tests:

`update_icons.dart` has a function checking that the new codepoints are a superset of the old ones. 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X ] I signed the [CLA].
- [ X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by JoseAlba — APPROVED** [review]
```
██╗      ██████╗ ████████╗███╗   ███╗
██║     ██╔════╝ ╚══██╔══╝████╗ ████║
██║     ██║  ███╗   ██║   ██╔████╔██║
██║     ██║   ██║   ██║   ██║╚██╔╝██║
███████╗╚██████╔╝   ██║   ██║ ╚═╝ ██║
╚══════╝ ╚═════╝    ╚═╝   ╚═╝     ╚═╝
```

File: dev/tools/update_icons.dart [hunk 1]
```diff
@@ -11,19 +11,24 @@ import 'dart:io';
...
+const String _newCodepointsPathOption = 'new-codepoints';
+const String _oldCodepointsPathOption = 'old-codepoints';
+const String _iconsClassPathOption = 'icons';
+const String _dryRunOption = 'dry-run';
 
-const String kDefaultCodepointsPath = 'bin/cache/artifacts/material_fonts/codepoints';
-const String kDefaultIconsPath = 'packages/flutter/lib/src/material/icons.dart';
+const String _defaultNewCodepointsPath = 'codepoints';
+const String _defaultOldCodepointsPath =
+    'bin/cache/artifacts/material_fonts/codepoints';
```
Comment (rami-a) [inline dev/tools/update_icons.dart:21]: Nit: This can all go on one line

File: dev/tools/update_icons.dart [hunk 2]
```diff
@@ -152,71 +156,143 @@ void main(List<String> args) {
...
-  argParser.addOption(kOptionIconsPath, defaultsTo: kDefaultIconsPath);
-  argParser.addFlag(kOptionDryRun, defaultsTo: false);
-  final ArgResults argResults = argParser.parse(args);
+  final ArgResults argResults = _handleArguments(args);
 
-  final File iconFile = File(path.absolute(argResults[kOptionIconsPath] as String));
-  if (!iconFile.existsSync()) {
-    stderr.writeln('Icons file not found: ${iconFile.path}');
+  final File iconClassFile = File(path
+      .normalize(path.absolute(argResults[_iconsClassPathOption] as String)));
```
Comment (rami-a) [inline dev/tools/update_icons.dart:162]: Nit: This can all go on one line

File: dev/tools/update_icons.dart [hunk 3]
```diff
@@ -152,71 +156,143 @@ void main(List<String> args) {
...
+  if (!iconClassFile.existsSync()) {
+    stderr.writeln('Error: Icons file not found: ${iconClassFile.path}');
+    exit(1);
+  }
+  final File newCodepointsFile = File(path.absolute(
+      path.normalize(argResults[_newCodepointsPathOption] as String)));
```
Comment (rami-a) [inline dev/tools/update_icons.dart:168]: Nit: This can all go on one line

File: dev/tools/update_icons.dart [hunk 4]
```diff
@@ -152,71 +156,143 @@ void main(List<String> args) {
...
+  if (!newCodepointsFile.existsSync()) {
+    stderr.writeln('Error: New codepoints file not found: ${newCodepointsFile.path}');
     exit(1);
   }
-  final File codepointsFile = File(path.absolute(argResults[kOptionCodepointsPath] as String));
-  if (!codepointsFile.existsSync()) {
-    stderr.writeln('Codepoints file not found: ${codepointsFile.path}');
+  final File oldCodepointsFile =
+      File(path.absolute(argResults[_oldCodepointsPathOption] as String));
```
Comment (rami-a) [inline dev/tools/update_icons.dart:174]: Nit: This can all go on one line

File: dev/tools/update_icons.dart [hunk 5]
```diff
@@ -152,71 +156,143 @@ void main(List<String> args) {
...
-  final String iconData = iconFile.readAsStringSync();
-  final String codepointData = codepointsFile.readAsStringSync();
-  final String newIconData = regenerateIconsFile(iconData, codepointData);
+  final String newCodepointsString = newCodepointsFile.readAsStringSync();
+  final Map<String, String> newTokenPairMap =
+      _stringToTokenPairMap(newCodepointsString);
+
+  final String oldCodepointsString = oldCodepointsFile.readAsStringSync();
+  final Map<String, String> oldTokenPairMap =
+      _stringToTokenPairMap(oldCodepointsString);
```
Comment (rami-a) [inline dev/tools/update_icons.dart:186]: Nit: This can all go on one line

File: dev/tools/update_icons.dart [hunk 6]
```diff
@@ -152,71 +156,143 @@ void main(List<String> args) {
...
 
-  if (argResults[kOptionDryRun] as bool)
+  _testIsMapSuperset(newTokenPairMap, oldTokenPairMap);
+
+  final String iconClassFileData = iconClassFile.readAsStringSync();
+
+  stderr.writeln('Generating new token pairs.');
+  final String newIconData =
+      _regenerateIconsFile(iconClassFileData, newTokenPairMap);
```
Comment (rami-a) [inline dev/tools/update_icons.dart:194]: Nit: This can all go on one line

File: dev/tools/update_icons.dart [hunk 7]
```diff
@@ -152,71 +156,143 @@ void main(List<String> args) {
...
 }
 
-String generateIconDeclarations(String codepointData) {
-  return LineSplitter.split(codepointData)
-      .map<String>((String l) => l.trim())
-      .where((String l) => l.isNotEmpty)
-      .map<String>(getIconDeclaration)
-      .join();
+void _testIsMapSuperset(
+    Map<String, String> newCodepoints, Map<String, String> oldCodepoints) {
```
Comment (rami-a) [inline dev/tools/update_icons.dart:257]: Nit: This can all go on one line

File: dev/tools/update_icons.dart [hunk 8]
```diff
@@ -152,71 +156,143 @@ void main(List<String> args) {
...
+      return rawIdentifier.replaceFirst(rewritePair.key, _identifierRewrites[rewritePair.key]);
+    }
+  }
+  return rawIdentifier;
+}
+
+// Replace the old codepoints file with the new.
+void _cleanUpFiles(File newCodepointsFile, File oldCodepointsFile) {
+  stderr
+      .writeln('\nMoving new codepoints file to ${oldCodepointsFile.path}.\n');
```
Comment (rami-a) [inline dev/tools/update_icons.dart:296]: Nit: This can all go on one line

**Review by rami-a — APPROVED** [review]
LGTM with a few nits.

**Comment by willlarche** [comment]
Having conversation around TTF/OTF. Do not merge yet.

**Review by dnfield — APPROVED** [review]
LGTM

File: dev/tools/update_icons.dart [hunk 9]
```diff
@@ -152,71 +155,135 @@ void main(List<String> args) {
...
+  stderr.writeln('Generating new token pairs.');
+  final String newIconData = _regenerateIconsFile(iconClassFileData, newTokenPairMap);
+
+  if (argResults[_dryRunOption] as bool) {
     stdout.writeln(newIconData);
-  else
-    iconFile.writeAsStringSync(newIconData);
+  } else {
+    stderr.writeln('\nWriting to ${iconClassFile.path}.');
+    iconClassFile.writeAsStringSync(newIconData);
```
Comment (HansMuller) [inline dev/tools/update_icons.dart:193]: If this fails, I assume we're OK with the default exception handler's report?

**Review by HansMuller — APPROVED** [review]
LGTM

File: dev/tools/update_icons.dart (hunk 9 above)
Comment (willlarche) [reply dev/tools/update_icons.dart:193]: Yeah. I don't think it can be improved.

**Comment by jesusrp98** [comment]
@willlarche what about the two-tone icon style? Is it implemented in this PR?

**Comment by willlarche** [comment]
Thanks @jessusrp98 ! Unfortunately the font we use for icons can only display 1 color. Two tone fonts have be imported as an image.

**Comment by jesusrp98** [comment]
Oh, I see. So does it mean that there are no further plans to implement that style as an option in the future?

Thanks!

**Comment by tvolkert** [comment]
> The wiki on updating material fonts needs to be updated

@willlarche did you make that update as well?

**Comment by rsheeter** [comment]
Re two-tone, I'd love to see it happen. Operating system and web browser support for color fonts has historically been a problem for color but this has improved somewhat lately; https://www.colorfonts.wtf/#section3.

Off the top of my head, to support twotone in Flutter we would need to:

- Confirm COLR (best UA support, we can compile two-tone to this format) works on all the environments Fluter supports
- Confirm `hb-subset` supports COLR and CPAL; implement these tables if not (my team can take care of that if necessary)
- Contemplate whether users should be able to override the color palette beyond just setting foreground color
   - This might mean we care about browser implementation of https://drafts.csswg.org/css-fonts-4/#font-palette-values

There are doubtless additional issues.

**Comment by willlarche** [comment]
@tvolkert Thanks for reminding me. Done!

**Comment by dragonblood** [comment]
so if I want to use rounded add_circle, how can i do that?

`icon: Icon(Icons.add_circle)`
this still displays filled icons

**Comment by willlarche** [comment]
Make sure you're on the master channel as this hasn't landed in stable yet.
I do see add_circle_rounded as an icon.

**Pull request merged**